src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaSecurity.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Channels;
+
+public sealed class KafkaSecurity
+{
+    internal const KafkaSecurityMode DefaultMode = KafkaSecurityMode.None;
+    private KafkaSecurityMode _mode;
+    private KafkaTransportSecurity _transportSecurity;
+    private KafkaMessageSecurity _messageSecurity;
+
+    public KafkaSecurity()
+        : this(DefaultMode, new KafkaTransportSecurity(), new KafkaMessageSecurity())
+    {
+    }
+
+    private KafkaSecurity(KafkaSecurityMode mode, KafkaTransportSecurity transportSecurity, KafkaMessageSecurity messageSecurity)
+    {
+        Fx.Assert(KafkaSecurityModeHelper.IsDefined(mode), $"Invalid SecurityMode value: {mode.ToString()}.");
+
+        if (!KafkaSecurityModeHelper.IsSupported(mode))
+        {
+            throw new NotSupportedException("Security mode not supported yet.");
+        }
+
+        _mode = mode;
+        _transportSecurity = transportSecurity ?? new KafkaTransportSecurity();
+        _messageSecurity = messageSecurity ?? new KafkaMessageSecurity();
+    }
+
+    [DefaultValue(DefaultMode)]
+    public KafkaSecurityMode Mode
+    {
+        get { return _mode; }
+        set
+        {
+            if (!KafkaSecurityModeHelper.IsDefined(value))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(value)));
+            }
+
+            if (!KafkaSecurityModeHelper.IsSupported(value))
+            {
+                throw new NotSupportedException("Security mode not supported yet.");
+            }
+
+            _mode = value;
+        }
+    }
+
+    public KafkaTransportSecurity Transport
+    {
+        get => _transportSecurity;
+        set => _transportSecurity = value ?? new KafkaTransportSecurity();
+    }
```

Comment:
This has the semantics of setting null sets it to defaults. This means the getter can return something different than was set. You could either throw for setting null, or in ApplySecurity do something like (_transportSecurity ?? new KafkaTransportSecurity()).ConfigureTransport....